src/coreclr/jit/hwintrinsicarm64.cpp

```diff
@@ -2222,9 +2222,8 @@ GenTree* Compiler::gtNewSimdConvertVectorToMaskNode(var_types   type,
     assert(varTypeIsSIMD(node));
 
     // ConvertVectorToMask uses cmpne which requires an embedded mask.
-    GenTree* embeddedMask = gtNewSimdHWIntrinsicNode(TYP_MASK, NI_Sve_CreateTrueMaskAll, simdBaseJitType, simdSize);
-    return gtNewSimdHWIntrinsicNode(TYP_MASK, embeddedMask, node, NI_Sve_ConvertVectorToMask, simdBaseJitType,
-                                    simdSize);
+    GenTree* trueMask = gtNewSimdAllTrueMaskNode(simdBaseJitType, simdSize);
+    return gtNewSimdHWIntrinsicNode(TYP_MASK, trueMask, node, NI_Sve_ConvertVectorToMask, simdBaseJitType, simdSize);
```

Comment:
Am I correct that unlike `general-purpose` and `SIMD/float` registers, predicate registers do not have a special concept to represent zero (select none/pfalse)? Likewise they don't have anything special to represent AllBitsSet (select all/ptrue) like x64 does?

So any instruction which requires a mask has to use an extra API to generate a `ptrue` instruction first?

Reply:
right, I don't think they have zero register or allset register.

Reply:
That's unfortunate, I hope that the JIT is able to generally do the right thing here and that we won't end up with a bunch of `ptrue` generated before instructions.

I know we have problems with generating `xorps` "too much" on x64 for example and have seen similar issues with `AllBitsSet` on both platforms (regardless of it being done in the front or back end).

Reply:
> I hope that the JIT is able to generally do the right thing here and that we won't end up with a bunch of `ptrue` generated before instructions.

I agree. Hopefully VN should eliminate bunch of those. We will see as we add more APIs.